feat: Allow timespan selection for group_by type dashboard chart

surajshetty3416 opened the pull request:
Allow timespan selection in "Group By" type of Dashboard Chart
<img width="1206" alt="Screenshot 2020-02-03 at 11 28 22 PM" src="https://user-images.githubusercontent.com/13928957/73677732-e069c900-46dc-11ea-9ee3-87c4051c301d.png">

surajshetty3416 committed:
feat: Allow timespan selection for group_by type dashboard chart

prssanna commented on frappe/desk/doctype/dashboard_chart/dashboard_chart.py:
```diff
@@ -86,23 +86,32 @@ def get_chart_config(chart, filters, timespan, timegrain, from_date, to_date):
+		to_date = datetime.datetime.now()
+
+	if not from_date:
+		from_date = get_from_date_from_timespan(to_date, timespan)
```
@surajshetty3416 Let's move this outside the function to avoid code duplication.

surajshetty3416 committed:
refactor: Commonify date range code

surajshetty3416 committed:
fix: Date range method

- No need to format date as it skips result records
that were created today after 00:00

prssanna commented on frappe/desk/doctype/dashboard_chart/dashboard_chart.py:
```diff
@@ -124,13 +124,13 @@ def get_group_by_chart_config(chart, filters, timespan, from_date, to_date):
 		return None
 
-def get_formatted_date_range(to_date, from_date, timespan):
+def get_date_range(to_date, from_date, timespan):
```
```suggestion
def get_date_range(from_date, to_date, timespan):
```

surajshetty3416 committed:
fix: Argument order

Co-Authored-By: Prssanna Desai <prssud@gmail.com>
